### DISCUSSION TITLE
Move exceeds maximum extrusion (1.203mm^2 vs 0.640mm^2) on print start

### DISCUSSION THREAD

**Discussion by Taff-The-Beard** [jschuh/klipper-macros#212] [Q&A] [answered]
Keep getting "Move exceeds maximum extrusion (1.203mm^2 vs 0.640mm^2)" on start of print just before the extruder goes for a purge line. I've doubled checked the start print/end print/ before and after layer change lines copied correctly in Super Slicer but no joy. 

I've read elsewhere about work around in disabling a saftey feature but don't wanna go down that route. 

Any ideas?

EDIT: I’ve removed the suggested start/end and layer change G-codes and it now runs. I’m still after a little help as to why these aren’t working for me.

[klippy (4).log](https://github.com/jschuh/klipper-macros/files/13692079/klippy.4.log)
[moonraker (1).log](https://github.com/jschuh/klipper-macros/files/13692080/moonraker.1.log)

**Comment by ovar0409** [comment]
I've got the same kind of error "Move exceeds maximum extrusion (1.804mm^2 vs 1.440mm^2)"... 1.44 because my nozzle is 0.6mm (Maximum Extrusion = 4.0 x Nozzle_Diameter^2).
I'm using  variable_start_purge_length: 30  and variable_start_purge_prime_length: 12.0, I guess it's not correct....

**Reply by ovar0409** [reply]
I increased variable_start_purge_length to  50 but it had no impact. I still get the exact same message "Move exceeds maximum extrusion (1.804mm^2 vs 1.440mm^2)"

**Comment by jschuh** [comment, marked as answer]
Just set `max_extrude_cross_section` to a more reasonable value in your Klipper config. You should be fine with 3.0, which is about twice Klipper's default for a 0.4mm nozzle. I personally set it to 4.0, but that's because I don't want to have update my Klipper config anytime I swap in a larger nozzle.

For a bit more context, Klipper's `max_extrude_cross_section` limit is intended to catch slicer errors (e.g. mismatching relative vs absolute extrusion modes), but it defaults to such a low value that it triggers in normal printing operations (priming/purge moves, taller layer heights, larger extrusion widths, certain arachne moves, etc.). Setting it a bit larger will still catch the real slicer errors, but you won't have prints spuriously aborting in normal operation.

**Reply by Taff-The-Beard** [reply]
Awesome sauce!!! Thank you so much! You’re a star!

**Reply by goegsig** [reply]
I got the same error. Thanks for the solution-

**Reply by ovar0409** [reply]
I read about that parameter, but I was not sure about the value to use.... and before changing it I wanted to understand a bit more. Thank you,
